app/js/enhancedDate.js

```diff
@@ -1,5 +1,107 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iffe() {
+  'use strict';
 
+  //variables for all of the functions below declared here
+
+
+  var myDate;
+  var myDayName;
+  var dayNames = ["Sunday", "Monday","Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+  var myMonth;
+  var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+  var todaysDate;
+
+
+  return {
+
+
+    setDate: function setDate(enterDate) {
+        if (enterDate instanceof Date){
+          myDate = enterDate;
+        } else if (isNaN(enterDate) === false){
+          myDate = enterDate;
+        } else{
+          myDate = new Date();
+        }
+    },
+
+
+    getDate: function getDate(enterDate) {
+         if (enterDate instanceof Date){
+            myDate = enterDate;
```

Comment:
enterDate will just be true or false, so rather than setting myDate to the true or false value, either set another variable, e.g. returnDate = myDate and return returnDate from the function, or just return myDate directly from this conditional block. I'll go over this in class and I think it will make more sense to see it used then. It's kind of unusual behavior for a function, so don't worry about getting tripped up by it.